pkg/orchestration/wiring/ec2compute/common/common.go

```diff
@@ -144,38 +187,69 @@ func WireUp(microServiceNameInSpec, ec2ComputePlanName string, stateResource *or
 		}
 
 		resourceReference := bindableShape.Data.ServiceInstanceName
-		bindingResources = append(bindingResources, wiringutil.ConsumerProducerServiceBinding(stateResource.Name, dependency.Name, resourceReference))
+		bindingResource := wiringutil.ConsumerProducerServiceBinding(stateResource.Name, dependency.Name, resourceReference)
+		bindingResources = append(bindingResources, bindingResource)
+		bindingResult = append(bindingResult, compute.BindingResult{
+			ResourceName:            dependency.Name,
+			BindableEnvVarShape:     bindableShape,
+			CreatedBindingFromShape: bindingResource,
+		})
 	}
 
-	dependencyReferences := make([]smith_v1.Reference, 0, len(bindingResources))
-	for _, res := range bindingResources {
-		dependencyReferences = append(dependencyReferences, smith_v1.Reference{
-			Resource: res.Name,
-		})
+	// filter out the list of dependencyReferences - this will be used
+	dependencyReferences := make([]smith_v1.Reference, 0, len(bindingResult))
+	shouldUseSecretPlugin := true
+	for _, res := range bindingResult {
+		ref := smith_v1.Reference{
+			Resource: res.CreatedBindingFromShape.Name,
+		}
+		dependencyReferences = append(dependencyReferences, ref)
+		if res.BindableEnvVarShape.Data.Vars == nil {
```

Comment:
Leave a TODO comment here too, saying that this is a temporary thing?

Reply:
I've got my next PR prepared and this will be gone